commit 91f04dfe1d93b8d2d383999ff5faf6411961caa1
Author: Ola x Nilsson <olani@axis.com>
Date:   Fri Mar 17 09:54:34 2023 +0100

    gnu: Handle basic file and time types for 32bit systems with time64
    
    32bit gnu types when _TIME_BITS=64 and _FILE_OFFSET_BITS=64

diff --git a/src/unix/linux_like/linux/gnu/b32/mod.rs b/src/unix/linux_like/linux/gnu/b32/mod.rs
index 54c84fa617..7cc9fb08e7 100644
--- a/src/unix/linux_like/linux/gnu/b32/mod.rs
+++ b/src/unix/linux_like/linux/gnu/b32/mod.rs
@@ -28,6 +28,16 @@ cfg_if! {
         pub type fsfilcnt_t = u64;
         pub type rlim_t = u64;
         pub type blksize_t = i64;
+    } else if #[cfg(gnu_time64_abi)] {
+        pub type time_t = i64;
+        pub type suseconds_t = i32;
+        pub type ino_t = u64;
+        pub type off_t = i64;
+        pub type blkcnt_t = i64;
+        pub type fsblkcnt_t = u64;
+        pub type fsfilcnt_t = u64;
+        pub type rlim_t = u64;
+        pub type blksize_t = i32;
     } else {
         pub type time_t = i32;
         pub type suseconds_t = i32;
